info/v1/container.go

```diff
@@ -266,25 +266,39 @@ type LoadStats struct {
 // CPU usage time statistics.
 type CpuUsage struct {
 	// Total CPU usage.
-	// Units: nanoseconds
+	// Unit: nanoseconds.
 	Total uint64 `json:"total"`
 
 	// Per CPU/core usage of the container.
 	// Unit: nanoseconds.
 	PerCpu []uint64 `json:"per_cpu_usage,omitempty"`
 
 	// Time spent in user space.
-	// Unit: nanoseconds
+	// Unit: nanoseconds.
 	User uint64 `json:"user"`
 
 	// Time spent in kernel space.
-	// Unit: nanoseconds
+	// Unit: nanoseconds.
 	System uint64 `json:"system"`
 }
 
+// Cpu Completely Fair Scheduler statistics.
+type CpuCFS struct {
+	// Total number of elapsed enforcement intervals.
+	Periods uint64 `json:"periods"`
+
+	// Total number of times tasks in the cgroup have been throttled.
+	ThrottledPeriods uint64 `json:"throttled_periods"`
+
+	// Total time duration for which tasks in the cgroup have been throttled.
+	// Unit: nanoseconds.
+	ThrottledTime uint64 `json:"throttled_time"`
+}
+
 // All CPU usage metrics are cumulative from the creation of the container
 type CpuStats struct {
 	Usage CpuUsage `json:"usage"`
+	CFS   CpuCFS   `json:"cfs"`
```

Comment:
The configured limits are already available under the spec namespace, see quota and period there. The resulting prometheus metrics can be seen here: https://github.com/google/cadvisor/blob/master/metrics/testdata/prometheus_metrics#L119-L124

The number of throttled periods and the total resulting throttled time already get exposed verbatim with the change. What exactly would you like to see?

I always prefer exporting metrics more or less directly. Most of the documentation users will find will discuss the standard cgroups metrics, I'd find it confusing to first read the code of cadvisor to understand how they were changed and exported instead.